compiler/src/main/kotlin/visitors/flowchecker.kt

```diff
@@ -0,0 +1,445 @@
+        )
+    }
+
+    override fun visit(node: ConditionalBlock): Flow {
+        visit(node.expr)
+        return visit(node.block)
+    }
+
+    override fun visit(node: BecomeStmt): Flow {
+        return visit(node.state)
+    }
+
+    //if a ReturnStmt is met, return true
+    override fun visit(node: ReturnStmt): Flow {
+        visit(node.value)
+        return Flow(containsReturn = true)
+    }
+
+    override fun visit(node: ForLoopStmt): Flow {
+        node.initPart?.let { visit(it) }
+        visit(node.condition)
+        node.postIterationPart?.let { visit(it) }
+        return visit(node.body)
+    }
+
+    //if a breakStmt is met, return true. containsBreak field represents both break and continue
+    override fun visit(node: BreakStmt): Flow {
+        // no-op
+        return Flow(containsBreak = true)
+    }
+
+    //if a ContinueStmt is met, return true. containsBreak field represents both break and continue
+    override fun visit(node: ContinueStmt): Flow {
+        // no-op
+        return Flow(containsBreak = true)
+    }
+
+    override fun visit(node: CodeBlock): Flow {
+        //variables used to
+        var containsReturn = false
+        var containsBreak = false
+
+        //visit all Stmts in the body, index used to print line number
+        for (i in 0 until node.body.size) {
+
+            val flow = visit(node.body[i])
+
+            //codeAfterThisBlock asserts that there is at least 2 Stmts in the body. No warnings will be
+            //produced if the current Stmt is the last, as it can not possibly block any further code inside the block
+            val codeAfterThisBlock = i < node.body.size - 1 || i == 0 && 1 < node.body.size
+
+            //saves the values from the visit()
+            if (flow.containsBreak) {
+                containsBreak = true
+            }
+
+            if (flow.containsReturn) {
+                containsReturn = true
+            }
+
+            /*makes sure that the current Stmt is not the last, else no Stmts will be blocked by a
+            return/break/continue*/
+            if (codeAfterThisBlock) when {
+                //if the visited node has a return that blocks further code in the block, produce warning
+                containsReturn -> {
+                    ErrorLogger += FlowWarning(
+                        node.body[i].ctx,
+                        "Code after statement never met, reason: return statement"
+                    )
+                    /*if a return blocking code is met, return. If this is the first Stmt in the codeblock
+                    the function still is guaranteed to meet a return, else it is not guaranteed to meet a return*/
+                    return if (i == 0) Flow(containsReturn = true) else Flow(containsReturn = false)
+                }
+                //if the visited node has a break/continue that blocks further code in the block, produce error
+                containsBreak && node.body[i] !is ForLoopStmt -> {
```

Comment:
This case is wrong. I can write:
```
if (x) {
    break;
} else {
    break;
}
```
and the result will be `Flow(false, false)` even though all branches contains a break. I will try to find a program, where this produces the wrong error...

Reply:
Found a function with wrong warnings:
```
function f2() int {

    for (let i = 0; false; i = i + 1) {
        if (true) { // No warning, but should give one!
            if (true) { // Warning: Code after statement never met, reason: break/continue statement
                break;
            } else {
                break;
            }

            let b = false;
        } else {
            if (true) { // Warning: Code after statement never met, reason: break/continue statement
                break;
            } else {
                break;
            }

            let b = false;
        }

        let reachable = false;
    }

    return 0;
}
```

Reply:
the returnvalue in the case that the stmt contains a break, is now set to Flow(containsReturn, containsBreak), so now the warning will flow up and produce more warnings.
commit e5f57895